Add documentation for custom resources, fix other docs as well (#112)
kubeshop/kusk-gateway: docs/customresources/envoyfleet.md

# Envoy Fleet

> This resource defines EnvoyFleet, which is the implementation of gateway in Kubernetes based on Envoy Proxy.

Reviewer: ```suggestion
This resource defines the EnvoyFleet, which is the implementation of the gateway in Kubernetes based on Envoy Proxy.
```

> Once resource manifest is deployed to Kubernets, it is used by Kusk Gateway Manager to setup K8s Envoy Proxy **Deployment** and **Service** with the type *LoadBalancer*.

Reviewer: ```suggestion
Once the resource manifest is deployed to Kubernetes, it is used by Kusk Gateway Manager to setup K8s Envoy Proxy **Deployment** and **Service** with the type *LoadBalancer*.
```

> Envoy Proxy is configured to connect for the configuration to [XDS](https://www.envoyproxy.io/docs/envoy/latest/api-docs/xds_protocol) service of KGW Manager.

Reviewer: ```suggestion
Envoy Proxy is configured to connect to the [XDS](https://www.envoyproxy.io/docs/envoy/latest/api-docs/xds_protocol) service of the KGW Manager to retrieve the configuration.
```

In its initial state there is no configuration, you have to deploy API or StaticRoute resource to setup the routing.

> If the Custom Resource is uninstalled, Manager deletes created K8s resouces.

Reviewer: ```suggestion
If the Custom Resource is uninstalled, the Manager deletes the created K8s resources.
```

> If the Custom Resource is uninstalled, Manager deletes created K8s resouces.

Reviewer 2: ```suggestion
If the Custom Resource Defintion is uninstalled, Manager deletes created K8s resources.
```

Currently supported parameters:

> * spec.**size** , which is the number of Envoy Proxy pods in K8s deployment, defaults to 1 if not specified.

Reviewer: ```suggestion
* spec.**size** , which is the number of Envoy Proxy pods in K8s the deployment, defaults to 1 if not specified.
```

> * metadata.**name** is the EnvoyFleet ID. Manager will supply the configuration for this specific ID.

Reviewer: ```suggestion
* metadata.**name** is the EnvoyFleet ID. The Manager will supply the configuration for this specific ID.
```

> * metadata.**name** is the EnvoyFleet ID. Manager will supply the configuration for this specific ID.

Reviewer 2: ```suggestion
* metadata.**name** is the EnvoyFleet ID. Manager will supply the configuration for this specific ID, matching the API / StaticRoute by fleet name. 
```

**Alpha Limitations**:

> * the support for multiple IDs configuration is a work in progress, right now used the single ID **default**.

Reviewer: ```suggestion
* the support for multiple IDs configuration is a work in progress, right now the static ID **default**. is used
```

> * the support for multiple IDs configuration is a work in progress, right now used the single ID **default**.

Reviewer 2: ```suggestion
* the support for multiple IDs configuration is a work in progress, right now Kusk Gateway Manager only uses the single ID **default**.
```

> * currently resource **status** field is not updated by manager when the configuration process finishes.

Reviewer: ```suggestion
* currently resource **status** field is not updated by the manager when the reconciliation of the configuration  finishes.
```
